addons/web/static/tests/core/misc_tests.js

```diff
@@ -0,0 +1,92 @@
+        setTimeout(resolve, 1);
+    });
+};
+
+QUnit.module('core', {
+    beforeEach: function () {
+        // We need to use a delay above 0 ms because otherwise the notification will close right after it opens
+        // before we can perform any test.
+        testUtils.mock.patch(Notification, {
+            _autoCloseDelay: 1,
+            _animation: false,
+        });
+    },
+    afterEach: function () {
+        // The Notification Service has a side effect: it adds a div inside
+        // document.body.  We could implement a cleanup mechanism for services,
+        // but this seems a little overkill since services are not supposed to
+        // be destroyed anyway.
+        $('.o_notification_manager').remove();
+        testUtils.mock.unpatch(Notification);
+    }
+}, function () {
+
+    QUnit.module('misc');
+
+    QUnit.test('test display_notification client action', async function (assert) {
+        assert.expect(6);
+
+        const ConcreteAction = AbstractAction.extend({
+            hasControlPanel: true,
+        });
+        core.action_registry.add('ConcreteAction', ConcreteAction);
+
+        const actionManager = await createActionManager({
+            services: {
+                notification: NotificationService,
+            },
+        });
+        await actionManager.doAction({
+            id: 1,
+            name: 'A Concrete Action',
+            tag: 'ConcreteAction',
+            type: 'ir.actions.client',
+        });
+
+        await actionManager.doAction({
+            type: 'ir.actions.client',
+            tag: 'display_notification',
+            params: {
+                title: 'title',
+                message: 'message',
+                sticky: false,
+            }
+        });
+
+        await testUtils.nextMicrotaskTick();
+        const $notification = $('body .o_notification_manager .o_notification');
+
+        assert.ok($notification.length > 0, "a notification should be present");
+
+
+        assert.strictEqual($notification.html().trim().replace(/\s+/g, ' '),
+            "<div class=\"toast-header\"> <span class=\"fa fa-2x mr-3 fa-lightbulb-o o_notification_icon\" role=\"img\" aria-label=\"Notification undefined\" title=\"Notification undefined\"></span> <div class=\"d-flex align-items-center mr-auto font-weight-bold o_notification_title\">title</div> </div> <div class=\"toast-body\"> <div class=\"o_notification_content\">message</div> </div>",
+            "should display notification");
+        assert.containsNone($notification, '.o_notification_close', "should not display the close button in ");
+        await waitCloseNotification();
+        assert.strictEqual($notification.is(':hidden'), true, "should hide the notification");
+        assert.strictEqual($('body .o_notification_manager .o_notification').length, 0, "should destroy the notification");
+
+        assert.strictEqual(actionManager.$('.breadcrumb').text(), 'A Concrete Action',
```

Comment:
check that there is a .o_kanban_view or whatever view type in the DOM (also, i'd do this test earlier, before triggering the client action)